applications/FluidDynamicsApplication/tests/cpp_tests/test_qs_vms_adjoint_element.cpp

```diff
@@ -0,0 +1,271 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Suneth Warnakulasuriya
+//
+
+// System includes
+#include <functional>
+
+// External includes
+
+// Project includes
+#include "containers/model.h"
+#include "testing/testing.h"
+#include "includes/constitutive_law.h"
+
+// Application includes
+#include "custom_utilities/fluid_adjoint_test_utilities.h"
+#include "custom_utilities/fluid_test_utilities.h"
+#include "includes/cfd_variables.h"
+#include "fluid_dynamics_application_variables.h"
+
+namespace Kratos
+{
+namespace Testing
+{
+
+namespace
+{
+
+ModelPart& CreateFluidQSVMSAdjointModelPart(
+    Model& rModel,
+    const std::string& rModelPartName,
+    const std::string& rElementName)
+{
+    const auto& set_variable_values = [](ModelPart& rModelPart) {
```

Comment:
Why creating this lambda function to be called in this same scope a few lines below. Wouldn't it be easier to first create the model part and then do all these calls with `r_model_part`?

Reply:
done :) [ Was copy pasting from other branch, and forgot... :D)